src/fides/api/alembic/migrations/versions/14acee6f5459_translation_data_clear_experience_configs.py

```diff
@@ -0,0 +1,565 @@
+"""translation data
+
+Revision ID: 14acee6f5459
+Revises: a1e23b70f2b2
+Create Date: 2024-01-09 21:17:13.115020
+
+"""
+
+import uuid
+from enum import Enum
+
+import yaml
+from alembic import op
+from loguru import logger
+from sqlalchemy import text
+from sqlalchemy.engine import ResultProxy
+
+# revision identifiers, used by Alembic.
+
+revision = "14acee6f5459"
+down_revision = "a1e23b70f2b2"
+branch_labels = None
+depends_on = None
+
+
+def generate_record_id(prefix):
+    return prefix + "_" + str(uuid.uuid4())
+
+
+class DefaultExperienceConfigTypes(Enum):
+    EEA_PRIVACY_CENTER = 1
+    EEA_MODAL_AND_BANNER = "pri-c8ff-78d6-4a02-850f-2c09dda-over-config"
+    EEA_TCF_OVERLAY = "d7c3ce0a-a3b3-43ff-bf6f-3d8-tcf-over-config"
+    US_MODAL = "pri-76b8-cc52-11ee-9eaa-8ef97a0-over-config"
+    US_PRIVACY_CENTER = "pri-27d4-cc53-11ee-9eaa-8ef97a04-pri-config"
+
+
+class ComponentType(Enum):
+    PrivacyCenter = "privacy_center"
+    Overlay = "overlay"
+    TCFOverlay = "tcf_overlay"
+
+
+EEA_MAPPING = {
+    ComponentType.PrivacyCenter: DefaultExperienceConfigTypes.EEA_PRIVACY_CENTER,
+    ComponentType.Overlay: DefaultExperienceConfigTypes.EEA_MODAL_AND_BANNER,
+    ComponentType.TCFOverlay: DefaultExperienceConfigTypes.EEA_TCF_OVERLAY,
```

Comment:
this is a good question around what's probably the most confusing thing about the script as written. i think you're pointing out a legitimate gap/bug.

basically, i tried to keep a "parallel" path for any given experience migration - if the experience hasn't had any manual updates that apply to it, then fall back to using the default template as-is. i think this should work for cases when there have been no changes to the TCF experience on a client env. but what it doesn't account for is when a client has changed the TCF experience with e.g. an update to a text field. i don't know if this has occurred (i can try to get a survey from the customer team), but it seems plausible, and it would cause a problem.

seems like it may just be worthwhile to build in some special-handling for TCF experience, since it (and only it) cannot actually be tied to notices, i.e. we can't derive its regions based on associated notices 👍

thanks for narrowing in on this!

Reply:
ok, i think i've covered now 👍